tests/test_ffmpeg.py

```diff
@@ -224,6 +226,44 @@ def test_write_not_contiguous(test_images):
             assert diff.mean() < 2.5
 
 
+def has_audio_streams(file_path):
+    command = ["ffprobe", "-show_streams", "-print_format", "json", file_path]
+    output = check_output(command)
+    parsed = json.loads(output)
+    streams = parsed["streams"]
+    audio_streams = list(filter((lambda x: x["codec_type"] == "audio"), streams))
+    return len(audio_streams) > 0
+
+
+def test_write_audio(test_images, tmp_path):
+    in_filename = test_images / "realshort.mp4"
+    out_filename = tmp_path / "realshort.mp4"
+    out_filename_audio = tmp_path / "realshort_audio.pm4"
+
+    in_file = []
+    with iio.read(in_filename, "ffmpeg") as R:
+        for i in range(10):
+            im = R.get_next_data()
+            in_file.append(im)
+
+    # Write without audio
+    with iio.save(out_filename, "ffmpeg") as W:
+        for im in in_file:
+            W.append_data(im)
+
+    # Now write with audio to preserve the audio track
+    with iio.save(
+        out_filename_audio, format="ffmpeg", audio_path=in_filename.as_posix()
+    ) as W:
+        for im in in_file:
+            W.append_data(im)
+
+    # Compare audio streams
+    assert has_audio_streams(in_filename)
```

Comment:
Considering that this is a constant and external to the test, it seems a bit redundant to make assertions about the input file. Can we remove this line?

Reply:
Sure, I like to add things like this to ensure that if for some reason whatever it's referencing changes, we'll know that the test is still valid. But I can see your point.

Reply:
```suggestion
```

Then it's a question of how much we trust the test data. It is certainly something we could test, but if we choose to do so, we should keep data tests separate from code tests. As such, I would like to remove it here.

Regarding data tests in general, we currently don't have them. We own the test data and it is very slow changing (I think the last actual addition/deletion was about 2 years ago) so not having tests for it has worked out thus far. That said, they certainly have their place, especially when you are testing with "live data" or external sources so I do like your thinking.